backend/fms_core/viewsets/_fetch_data.py

```diff
@@ -0,0 +1,133 @@
+        .annotate(derived_count=Count("derived_samples"))
+        .annotate(extracted_from=Subquery(
+                SampleLineage.objects
+                .filter(child=OuterRef("pk"), process_measurement__process__protocol__name="Extraction")
+                .values_list("parent", flat=True)[:1]
+            )
+        )[offset:offset+limit]
+        .values(
+            'id',
+            'name',
+            'container_id',
+            'coordinates',
+            'volume',
+            'concentration',
+            'creation_date',
+            'quality_flag',
+            'quantity_flag',
+            'depleted',
+            'comment',
+            'is_library',
+            'derived_count',
+            'first_derived_sample',
+            'derived_samples__project',
+            'child_of',
+            'extracted_from',
+            'process_measurement',
+            'created_by',
+            'created_at',
+            'updated_by',
+            'updated_at',
+            'deleted',
+        )
+    )
+    samples = { s["id"]: s for s in sample_values_queryset }
+    derived_sample_ids = [sample_values["first_derived_sample"]for sample_values in sample_values_queryset]
+    derived_sample_values_queryset = (
+        DerivedSample.objects
+        .filter(id__in=derived_sample_ids)
+        .values(
+            "id",
+            "biosample_id",
+            "biosample__alias",
+            "sample_kind_id",
+            "tissue_source_id",
+            "biosample__collection_site",
+            "experimental_group",
+            "biosample__individual_id",
+        )
+    )
+    derived_samples = { ds["id"]: ds for ds in derived_sample_values_queryset }
+
+    serialized_data = []
+    for sample in samples.values():
+        derived_sample = derived_samples[sample["first_derived_sample"]]
+        is_pool = sample["derived_count"] > 1
+        is_library = sample["is_library"]
+        data = {
+            'id': sample["id"],
+            'biosample_id': derived_sample["biosample_id"] if not is_pool else None,
+            'name': sample["name"],
+            'alias': derived_sample["biosample__alias"] if not is_pool else None,
+            'volume': sample["volume"],
+            'depleted': sample["depleted"],
+            'concentration': sample["concentration"],
+            'child_of': sample["child_of"],
+            'extracted_from': sample["extracted_from"] if not is_pool else None,
+            'individual': derived_sample["biosample__individual_id"] if not is_pool else None,
+            'container': sample["container_id"],
+            'coordinates': sample["coordinates"],
+            'sample_kind': derived_sample["sample_kind_id"] if not is_pool or (is_pool and not is_library) else "POOL",
+            'is_library': is_library,
+            'is_pool': is_pool,
+            'projects': sample["derived_samples__project"],
```

Comment:
Might be better to change to 'project'

Reply:
There is still many projects by sample potentially with the pools.